src/components/RecipeSubmit.vue

```diff
@@ -0,0 +1,297 @@
+                              hint="Enter a piece of equipment (e.g. &quot;Rolling pan&quot;) and press Tab to record it." persistent-hint>
+                        </v-select>
+
+                        <!-- TODO: Tags -->
+
+                    </md-layout>
+                </md-layout>
+                <md-button type="submit" class="md-raised md-primary">Submit Recipe</md-button>
+            </form>
+        </div>
+    </div>
+    </v-app>
+</template>
+
+<script>
+  import Firebase from 'firebase';
+  import db from '../database';
+  import router from 'vue-router';
+  import uuidV4 from 'uuid/v4';
+  import VSelect from "vuetify/es5/components/VSelect/VSelect";
+  import MdLayout from "../../node_modules/vue-material/src/components/mdLayout/mdLayout.vue";
+  import MdInput from "../../node_modules/vue-material/src/components/mdInputContainer/mdInput.vue";
+  import VTextField from "vuetify/es5/components/VTextField/VTextField";
+
+  export default {
+    components: {
+      VTextField,
+      VSelect,
+      MdLayout
+    },
+    name: 'recipeSubmit',
+    data () {
+      return {
+        showCarousel: true,
+        localPhotos: [], // Photos stored in the browser
+
+        recipeName: undefined,
+        recipeDescription: undefined,
+        ingredients: [],
+        ingredientOptions: [],
+        prepTimeHr: undefined,
+        prepTimeMin: undefined,
+        cookTimeHr: undefined,
+        cookTimeMin: undefined,
+        servings: undefined,
+        cookware: [],
+        cookwareOptions: [],
+        photos: [],
+        tags: [],
+        recipeSteps: [
+          {}
+        ]
+      };
+    },
+    mounted: function() {
+      // Load cookware dropdown options
+      var cookware = [];
+      db.ref('cookware/').orderByChild('description').on('child_added', function(c) {
+        cookware.push(c.val().description);
+      });
+      this.cookwareOptions = cookware;
+
+      // Load ingredient dropdown options
+      var ingredients = [];
+      db.ref('ingredients/').orderByChild('name').on('child_added', function(i) {
+        ingredients.push(i.val().name);
+      });
+      this.ingredientOptions = ingredients;
+    },
+    methods: {
+      onFileChange(e) {
+        this.showCarousel = false; // Reinitialization hack because dynamic images won't be supported until v1.0
+
+        var files = e.target.files || e.dataTransfer.files;
+        if (!files.length) {
+          return;
+        }
+
+        var storageRef = Firebase.storage().ref(); // TODO: How do I put this in database.js like we did with db?
+
+        var fileList = [];
+        var photoList = [];
+
+        for (var i = 0; i < files.length; i++) {
+          fileList.push({src: window.URL.createObjectURL(files[i])});
+
+          var uploadTask = storageRef.child('images/recipes/' + uuidV4()).put(files[i]);
+          uploadTask.on('state_changed', function(snapshot) {
```

Comment:
Prefer ES6 arrow functions when value of 'this' does not matter. Ex:
uploadTask.on('state_changed', (snapshot) => {